src/java/org/apache/cassandra/config/Config.java

```diff
@@ -216,38 +250,43 @@
+    @Replaces(oldName = "batch_size_warn_threshold_in_kb", converter = Converters.KIBIBYTES_DATASTORAGE, deprecated = true)
+    public volatile DataStorageSpec batch_size_warn_threshold = new DataStorageSpec("5KiB");
+    @Replaces(oldName = "batch_size_fail_threshold_in_kb", converter = Converters.KIBIBYTES_DATASTORAGE, deprecated = true)
+    public volatile DataStorageSpec batch_size_fail_threshold = new DataStorageSpec("50KiB");
     public Integer unlogged_batch_across_partitions_warn_threshold = 10;
     public volatile Integer concurrent_compactors;
-    public volatile int compaction_throughput_mb_per_sec = 16;
-    public volatile int compaction_large_partition_warning_threshold_mb = 100;
-    public int min_free_space_per_drive_in_mb = 50;
+    //follow up on the below one Kate!
+    @Replaces(oldName = "compaction_throughput_mb_per_sec", converter = Converters.MEBIBYTES_PER_SECOND_DATA_RATE, deprecated = true)
+    public volatile DataRateSpec compaction_throughput = new DataRateSpec("16MiB/s");
+    @Replaces(oldName = "compaction_large_partition_warning_threshold_mb", converter = Converters.MEBIBYTES_DATA_STORAGE, deprecated = true)
+    public volatile DataStorageSpec compaction_large_partition_warning_threshold = new DataStorageSpec("100MiB");
+    @Replaces(oldName = "min_free_space_per_drive_in_mb", converter = Converters.MEBIBYTES_DATA_STORAGE, deprecated = true)
+    public DataStorageSpec min_free_space_per_drive = new DataStorageSpec("50MiB");
     public volatile Integer compaction_tombstone_warning_threshold = 100000;
 
     public volatile int concurrent_materialized_view_builders = 1;
     public volatile int reject_repair_compaction_threshold = Integer.MAX_VALUE;
+    
+    @Replaces(oldName = "stream_throughput_outbound_megabits_per_sec", converter = Converters.MEGABITS_TO_MEBIBYTES_PER_SECOND_DATA_RATE, deprecated = true)
+    public volatile DataRateSpec stream_throughput_outbound = new DataRateSpec("24MiB/s");
+    @Replaces(oldName = "inter_dc_stream_throughput_outbound_megabits_per_sec", converter = Converters.MEGABITS_TO_MEBIBYTES_PER_SECOND_DATA_RATE, deprecated = true)
+    public volatile DataRateSpec inter_dc_stream_throughput_outbound = new DataRateSpec("24MiB/s");
 
-    /**
-     * @deprecated retry support removed on CASSANDRA-10992
-     */
-    @Deprecated
-    public int max_streaming_retries = 3;
-
-    public volatile int stream_throughput_outbound_megabits_per_sec = 200;
-    public volatile int inter_dc_stream_throughput_outbound_megabits_per_sec = 200;
-
-    public volatile int entire_sstable_stream_throughput_outbound_megabits_per_sec = 200;
-    public volatile int entire_sstable_inter_dc_stream_throughput_outbound_megabits_per_sec = 200;
+    public DataRateSpec entire_sstable_stream_throughput_outbound = new DataRateSpec("25MiB/s");
+    public DataRateSpec entire_sstable_inter_dc_stream_throughput_outbound = new DataRateSpec("25MiB/s");
```

Comment:
Should these remain `volatile`, given they are still settable via JMX?

Reply:
Definitely we shouldn't drop it, this change was not intentional. Thank you!